### PR TITLE
[SandboxIR] Add ExtractValueInst.

### PR DESCRIPTION
**Pull request by slackito**

### CODE REVIEW

File: llvm/include/llvm/SandboxIR/SandboxIR.h [hunk 1]
```diff
@@ -1621,6 +1626,61 @@ class UnaryInstruction
...
+public:
+  static Value *create(Value *Agg, ArrayRef<unsigned> Idxs, BBIterator WhereIt,
+                       BasicBlock *WhereBB, Context &Ctx,
+                       const Twine &Name = "");
+
+  static bool classof(const Value *From) {
+    return From->getSubclassID() == ClassID::ExtractValue;
+  }
+
+  static Type *getIndexedType(Type *Agg, ArrayRef<unsigned> Idxs) {
```
Comment (vporpo) [inline llvm/include/llvm/SandboxIR/SandboxIR.h:1645]: nit: It is not straight forward what this function does, so I would copy LLVM IR's comments.

File: llvm/unittests/SandboxIR/SandboxIRTest.cpp [hunk 2]
```diff
@@ -1261,6 +1261,102 @@ define void @foo(<2 x i8> %v1, <2 x i8> %v2) {
...
+
+  auto *Zero = sandboxir::ConstantInt::get(Type::getInt32Ty(C), 0, Ctx);
+  EXPECT_EQ(ShouldBeConstant, Zero);
+
+  // getIndexedType
+  Type *AggType = ExtNested->getAggregateOperand()->getType();
+  EXPECT_EQ(sandboxir::ExtractValueInst::getIndexedType(
+                AggType, ArrayRef<unsigned>({1, 0})),
+            llvm::ExtractValueInst::getIndexedType(AggType,
+                                                   ArrayRef<unsigned>({1, 0})));
```
Comment (vporpo) [inline llvm/unittests/SandboxIR/SandboxIRTest.cpp:1318]: Could you have two checks for `getIndexedType()`, one that returns non-null and one that returns null. The reason is that once we migrate to `sandboxir::Type` the null case would need special treatment.

File: llvm/include/llvm/SandboxIR/SandboxIR.h (hunk 1 above)
Comment (slackito) [reply llvm/include/llvm/SandboxIR/SandboxIR.h:1649]: Done.

File: llvm/unittests/SandboxIR/SandboxIRTest.cpp (hunk 2 above)
Comment (slackito) [reply llvm/unittests/SandboxIR/SandboxIRTest.cpp:1318]: Added a simple test for the nullptr case. Please take another look.

**Review by vporpo — APPROVED** [review]
LG thanks!

**Pull request merged**